hooks/useOnClickAway.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, RefObject } from 'react';
+
+const useOnClickAway = (ref: RefObject<Node>, action: () => void): void => {
+  useEffect(() => {
+    const handleAction = (event: MouseEvent | TouchEvent) => {
+      if (!ref.current || ref.current.contains(event.target as Node)) return;
+      action();
+    };
+
+    document.addEventListener('mousedown', handleAction);
+    document.addEventListener('touchstart', handleAction);
```

Comment:
Isn't it true that `mousedown` pretty much equals `touchstart` on mobile devices? If I am right, we don't have to handle `touchstart` - `mousedown` should be enough.

Reply:
Maybe you are right. That additional eventListener is form over substance and isn't required in this case - just listen on mousedown event is enough.